Week3/homework/step2-7.js

```diff
@@ -20,4 +20,12 @@ f2(y);
 
 console.log(y);
 
-// Add your explanation as a comment here
+// The first function incriminate variable of val 1 number.
+// Then returns the number as a variable.
+// In the console it is written 9, because x was a const variable.
+// That means x never changes.
+
+// The second function incriminate x property of an object called val.
+// Then returns the number as an object.
+// In the console it is written 10, because y was a const object.
+// That means y doesn't change, but the content of y can change.
```

Comment:
continuing the previous comment but this time you have hitted the spot.
> The second function incriminate x property of an object called val.
> Then returns the number as an object.
> In the console it is written 10, because y was a const object.
> That means y doesn't change, but the content of y can change.

to be more accurate, an object is just a reference to what it contains, that means the `y` is just somewhere else in our computer, and that makes it changes.

just by passing the reference to `val` that doesn't means `val` contains the exact thing as `x`, but all it contains is a reference that points to where `x` is located.

**And this is the difference between `passing by reference` and `passing by value`**
A value never change unless you change it directly by it's variable name
A reference points to it's content on the desk and once that changed somehow it changes everywhere.